src/Features/CSharp/Portable/CodeFixes/ConditionalExpressionInStringInterpolation/AddParenthesisCodeAction.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.ConditionalExpressionInStringInterpolation
+{
+    internal partial class CSharpAddParenthesisAroundConditionalExpressionInInterpolatedStringCodeFixProvider
+    {
+        private sealed class AddParenthesisCodeAction : CodeAction
+        {
+            public AddParenthesisCodeAction(Document document, int conditionalExpressionSyntaxStartPosition)
+            {
+                this.Document = document;
+                this.ConditionalExpressionSyntaxStartPosition = conditionalExpressionSyntaxStartPosition;
+            }
+
+            private Document Document { get; }
+            private int ConditionalExpressionSyntaxStartPosition { get; }
+
+            public override string Title => CSharpFeaturesResources.AddParenthesisAroundConditionalExpressionInInterpolatedString;
+
+            protected override async Task<Document> GetChangedDocumentAsync(CancellationToken cancellationToken)
+            {
+                var text = await Document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                var openParenthesisPosition = ConditionalExpressionSyntaxStartPosition;
+                var textWithOpenParenthesis = text.Replace(openParenthesisPosition, 0, "(");
```

Comment:
it woudl be more appropriate to just fix up teh syntax tree, rather than manually making text edits.  You could just get the InterpolationSyntax (that you got back when you did hte analysis) and wrap it's contained expression with a ParenthesizedExpression.

Reply:
That is what I had [previously](https://github.com/dotnet/roslyn/pull/23587/commits/d0bdaf769bb4ea57a26a0be8e326bd41f051eb1e#diff-688e183cf41536b4fde1bbb91af4224fR27). See commit d0bdaf769bb4ea57a26a0be8e326bd41f051eb1. This worked well in many cases but was problematic for string literals in the false condition because the resulting syntax trees are bogus and hard to transform. So @sharwell and I decided to give this unusal approach a try (See also comments above).

Reply:
Gotcha!  Ok.  Let's go with this approach then :)